### PR TITLE
feat: changed UX of download and delete action buttons on widget hover and selected state

### PR DESCRIPTION
**Pull request by Kausar-HM**
This PR includes the changes for UX of download and delete buttons on widget hover and selected state #2439 
Please note this is first PR related to ticket #2439 , for changes related to title, we will be having another PR.

Verifying changes:

Before changing UX (video taken on today's latest rc):
[Screencast from 19-01-24 06:52:01 PM IST.webm](https://github.com/awslabs/iot-app-kit/assets/154328164/4d4176e0-fdfd-4a0f-aefe-6840fde4dcd0)

After changes in the PR:
[Screencast from 19-01-24 06:49:13 PM IST.webm](https://github.com/awslabs/iot-app-kit/assets/154328164/7cf8d7d3-bb54-4785-9bbd-06139b0de335)

### CODE REVIEW

File: packages/dashboard/src/components/widgets/tile/tile.css [hunk 1]
```diff
@@ -26,6 +27,16 @@
...
-.tile-button-contianer {
+.tile-button-container {
   display: flex;
-}
+  align-items: center;
+  position: absolute;
+  right: 19px;
+  top: -20px;
+  height: 24px;
+  padding: 4px 9px;
```
Comment (Chandru-HM) [inline packages/dashboard/src/components/widgets/tile/tile.css:37]: can you please use cloudscape design tokens here

File: packages/dashboard/src/components/widgets/tile/tile.tsx [hunk 2]
```diff
@@ -43,7 +46,7 @@ const DeletableTileAction = ({ handleDelete }: DeletableTileActionProps) => {
       <Button
         onClick={handleDelete}
         ariaLabel='delete widget'
-        variant='icon'
+        variant={variant ? variant : 'icon'}
```
Comment (gmuralidharreddy) [inline packages/dashboard/src/components/widgets/tile/tile.tsx:49]: Same here,
use variant ?? 'icon'

File: packages/dashboard/src/components/widgets/tile/tile.css [hunk 3]
```diff
@@ -26,6 +27,16 @@
...
+  border: 2px solid #0972D3;
+  background-color: #FFF;
+  border-radius: 5px;
```
Comment (Chandru-HM) [inline packages/dashboard/src/components/widgets/tile/tile.css:40]: can you please use cloudscape design tokens here

File: packages/dashboard/src/components/csvDownloadButton/index.tsx [hunk 4]
```diff
@@ -131,7 +132,7 @@ export const CSVDownloadButton = ({
       ariaLabel='download CSV'
       iconName='download'
       loading={isDownloading}
-      variant='icon'
+      variant={variant ? variant : 'icon'}
```
Comment (gmuralidharreddy) [inline packages/dashboard/src/components/csvDownloadButton/index.tsx:135]: you can also use like below,
variant ?? 'icon'

**Review by gmuralidharreddy — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/components/widgets/tile/tile.tsx [hunk 5]
```diff
@@ -104,6 +104,18 @@ const WidgetTile: React.FC<WidgetTileProps> = ({
     });
   };
 
+  const handleOnMouseLeave = () => {
+    if (selectedWidgets && selectedWidgets.find((w) => w.id === widget.id))
+      setShowActionButtons(true);
+    else setShowActionButtons(false);
```
Comment (gmuralidharreddy) [inline packages/dashboard/src/components/widgets/tile/tile.tsx:110]: You can create a function and use it handleMouseLeave and useEffect

File: packages/dashboard/src/components/widgets/tile/tile.tsx [hunk 6]
```diff
@@ -113,56 +125,62 @@ const WidgetTile: React.FC<WidgetTileProps> = ({
...
+        </div>
+      )}
+
       {headerVisible && (
         <div
           style={{
             display: 'flex',
             justifyContent: 'space-between',
             padding: `${spaceScaledXs} ${spaceScaledXs} ${spaceScaledXxs} ${spaceScaledM}`,
-            borderBottom: `2px solid ${colorBorderDividerDefault}`,
```
Comment (gmuralidharreddy) [inline packages/dashboard/src/components/widgets/tile/tile.tsx:123]: Better to add this borderBottom until we completely remove the header and show the title in respective charts.

File: packages/dashboard/src/customization/widgets/text/component.tsx [hunk 7]
```diff
@@ -53,13 +54,17 @@ const TextWidgetComponent: React.FC<TextWidget> = (widget) => {
...
-    }
+    return (
+      <WidgetTile widget={widget}>
+        {isUrl ? (
+          <StyledTextArea isUrl {...props} />
+        ) : isEditing ? (
+          <StyledTextArea {...props} />
+        ) : (
+          <EditableStyledText {...props} />
+        )}
```
Comment (gmuralidharreddy) [inline packages/dashboard/src/customization/widgets/text/component.tsx:65]: Avoid using multiple ternary operators in single statements, you can add tiny component or function and return like below:

```
const sampleFunc = ({isUrl, isEditing}) => {
if(isUrl) {
return <StyledTextArea isUrl {...props} />;
} else if(isEditing) {
return <StyledTextArea {...props} />;
} else {
return <EditableStyledText {...props} />;
}
```

**Comment by Chandru-HM** [comment]
mouse hover is not showing delete icon or highlight the other widget on the Bar widget, please take a look
[widget-not-highlight.webm](https://github.com/awslabs/iot-app-kit/assets/142866907/e6701a3b-8340-416f-86a6-c84bb6787d79)

**Comment by Chandru-HM** [comment]
Table header is hiding the other widgets, please take a look
[table-header-issue.webm](https://github.com/awslabs/iot-app-kit/assets/142866907/73055799-eec6-487c-b0cd-c7a84b657630)

**Pull request closed without merging**